### DISCUSSION TITLE
Announcements, team updates, the future, and more!

### DISCUSSION THREAD

**Discussion by TurboTurtle** [sosreport/sos#3466] [Announcements]
Hi all. It has been a while since the last announcement within the project, but that doesn't mean things haven't been moving. I'd like to take a moment to go over the current state of sos, make some announcements, and generally get rolling on 2024 changes for the project.

### GitHub Discussions

As you might have noticed, this is posted to GH discussions - and is the first such post for the sosreport organization (well...technically? We had some earlier that seem to have been taken behind the barn when discussions went under an overhaul for GitHub a while back).

We'd like to see more communication within the project going forward. We have several such avenues already - e.g. IRC and the mailing list - but those inevitably go silent. I'd like to invite all of you to engage on GH to spark new ideas and conversations in perhaps a more asynchronously-friendly medium, here on the discussions board.

### Team Updates

I'd like to talk about 2 specific changes for the make up of the sos team.

#### A new maintainer!

First and foremost - the maintainers team. For quite a while now this has been myself and Pavel, and while sos is not largest project out there, it often helps to have more than 2 people taking care of the day-to-day work that comprises maintaining an open source project.

That said, sos is still an extremely useful and important tool for the audience and use cases we develop and make releases for. So it is critically important that whomever dons the mantle of "sos maintainer" be technically competent, have a firm grasp and understanding of the aim of the project, and have a genuine desire in its success, stability, and future state.

So, after a lot of thought and consideration I am happy to welcome Arif Ali (@arif-ali) to the maintainer's team.

Arif has been a consistent contributor to sos and has made significant contributions to ensuring its success and suitability for Ubuntu, and I look forward to seeing how that particular downstream evolves from an sos perspective going forward.

#### Spring cleaning, and the overall team makeup

We've had a lot of contributors over the years to the project, and a handful of which have joined the sos team as collaborating members. This is great, but we're now at a point where the majority of the team makeup are not active and have not been for some time (in most cases not just for sos, but inactive on GH entirely).

There are two asks I have regarding team membership:

1. **Everyone who wishes to remain on the team please respond to this post**. A non-response will imply to me that you are not interested in continuing, which is perfectly fine. I'm not looking to "cut down membership" or anything like that (see ask 2 below), but I feel that a team that is _mostly_ active tends to snowball itself into a more engaged team among active members. Further, I'd like to explore the possibility of expanding the role assignments (e.g. reviewers, code owners, etc...) with the team, and it makes the most sense to do so with an active pool of participants.
2. If you'd like to onboard new people to the team - contributors, testers, project managers, etc... by all means please contact one of the maintainers - that being myself, Pavel (@pmoravec), and Arif (@arif-ali). We can all add folks and organize into relevant teams and hopefully flesh out some ideas on where to take the project next. We've expanded our "reach" as it were over the last 3 or so years to include more distributions than ever before, and it'd be helpful to have representation from those downstreams to keep us clued into how the project is being used across different environments.

### Roadmapping

We're coming up on the 4.7.0 release in February, with our regular cadence release six months later in August for 4.8.0. As part of the desire to foster more discussion here on GH, I'd like to ask for anyone to suggest a focus for the next development cycle starting in February. Have an idea for a new subcommand? A refactor of how we do something today? Want to explore a new use case for sos? Got an idea of how to fix translations? Let us know (especially that last one! ;) ), and we can carve new paths for the project!

I'm happy to be a part of the project and what we've been able to accomplish to date, and I can't wait to see how sos evolves over the next months and years.

**Comment by jjansky1** [comment]
I will like to remain on the team.

**Comment by bmr-cymru** [comment]
I'd like to stick around

**Comment by pmoravec** [comment]
Same here. I am glad to contribute to these roles, at least:

- review PRs
- (raise and) discuss new ideas, features etc.
- actively contribute to the code (something that is bit declining over the time, though)
- merging approved PRs, preparing new release etc. (something I feel more as a duty than a pleasure, though)

I feel the project deserves below activities more than what is happening now, but I dont have bandwidth or skills or capabilities:

- put more focus on (avocado) tests as part of PRs; we rarely add a test even when adding a feature that deserves a test coverage (and I am an expert in forgetting this!)
- review old issues or PRs that are idle for some time (6months? 1year?)
- have one place with longer-term goals or pains (to list there e.g. translations, "sos upload" feature, concurrent cleaner,..)
- some roadmap / visionary approach can be beneficial as well..?

**Reply by arif-ali** [reply]
The folks at Canonical (me and @dnegreira) are fully on board on the testing bits.

Over the past year, we have been a lot more active, and have needed to delay or abandon deb releases due to internal testing not going to plan, and it was not being testing here as an upstream project (we have then added the relevant fixes ;)). 

We are hoping to dedicate some time on some of the testing, so that more things are automated, and less manual things have to be done to accomplish releases.

I think issue #3467 is a prime example of something we could have seen earlier and resolved, and some of the prior py310 and py311 issues we had faced based on deprecations.

**Comment by mhradile** [comment]
I'm honestly not sure if me and @tiredPotato need to be on the team. We need this mostly to be able to give approvals on selected PRs. So if active participation is required for membership we will have to do it outside of github.

Being members just sounded most transparent and easy way to do this.

**Reply by TurboTurtle** [reply]
I don't have any plans for measuring activity for membership or anything - again this "raise your hand" request is just to level set.

Going forward, if you want and/or need to be on the team even just for weighing in on select PRs, that's fine.

**Comment by dnegreira** [comment]
Hi, I would like to be part of the team, even though over the last few months (years?) I have not been very active in the development of sos, hopefully that will change starting this year as I will start to take a more active role in sos under the Ubuntu/Canonical umbrella.

I am looking to take ownership of the downstream Ubuntu packaging, as well as contribute with new plugins and reviews of code in sos where necessary.
I am also looking forward to be able to help with preparing new releases to relief a bit the burden of the maintainers, contribute to discussions and expand the testing coverage, specially for the Ubuntu bits, but happy to help with the overall improvement of tests.

**Comment by jcastill** [comment]
I'd like to remain on the team.

**Comment by tiredPotato** [comment]
I would like to stay on the team .

**Comment by bvassova** [comment]
Hi, I'd like to continue too

**Comment by TurboTurtle** [comment]
Last call - I'll be swinging through the membership list at the end of the week.
